src/back/glsl/mod.rs

```diff
@@ -2455,7 +2457,7 @@ impl<'a, W: Write> Writer<'a, W> {
             match self.module.types[var.ty].inner {
                 crate::TypeInner::Struct { .. } => match var.class {
                     crate::StorageClass::Uniform | crate::StorageClass::Storage => {
-                        let name = self.reflection_names[&var.ty].clone();
+                        let name = self.reflection_names_uniforms[&var.ty].clone();
```

Comment:
why are we looking up `var.ty` instead of looking up `handle` directly in that other map?

Reply:
I believe this is because the `write_struct` function, where [`reflection_names_uniforms.insert` is called](https://github.com/gfx-rs/naga/blob/ab2fa435e6c7e3f0026cb753ddcf1a4a16dcee7e/src/back/glsl/mod.rs#L1241), has no access to the global variable handle, only the type handle.

Reply:
Yep, we need a block struct name, that will be generated as uniform buffer. 

https://github.com/gfx-rs/naga/blob/ab2fa435e6c7e3f0026cb753ddcf1a4a16dcee7e/src/back/glsl/mod.rs#L1233-L1239

Reply:
If we will use the global variable handle here, we still need to check the variable type and ensure that the variable is a struct with a uniform storage class.

If we want to change this, we need to change block struct writing.

Reply:
ok I see. So the block name shows up in the API, not the variable name, and the block name comes from the type. Ok